### ISSUE TITLE
Search input: focus state style (Desktop)

### ISSUE THREAD

**Issue by davidhunter08** [nhsuk/nhsuk-frontend#679]
## What

A recent accessibility audit has flagged that the search input visible focus indicator did not meet the minimum contrast ratio of 3:1.

![Screenshot 2021-01-07 at 15 21 04](https://user-images.githubusercontent.com/14331000/103998020-97937d80-5193-11eb-9025-074477aed4db.png)

> The colour contrast between the highlighting against the white background fails to meet the required ratio of 3.1:
> 
> This is consistent wherever the yellow highlighting is used on a white background.

**WCAG Reference:**
1.4.3 Contrast (Minimum) (Level AA)
Understanding Contrast (Minimum) https://www.w3.org/WAI/WCAG21/Understanding/contrast-minimum.html |How to Meet Contrast (Minimum) https://www.w3.org/WAI/WCAG21/quickref/#contrast-minimum

**Comment by davidhunter08** [comment]
My understanding was that the ‘background’ in this context is the blue and not white, therefore it meets the required ratio of 3.1.

GOV have similar implementations on GOV.UK and Design System so might have found the same issue?

![govuk-search](https://user-images.githubusercontent.com/14331000/103999276-9878df00-5194-11eb-982f-29ec71fdc453.gif)

![govuk-ds-search](https://user-images.githubusercontent.com/14331000/103999284-9b73cf80-5194-11eb-83e8-dfcf325ab10b.gif)

We should also check if the NHS logo fails to meet the minimum contrast ratio with the colours being the same (white object, yellow focus, blue background), but this issue was not raised in the audit.

<img width="128" alt="Screenshot 2021-01-07 at 17 35 02" src="https://user-images.githubusercontent.com/14331000/103999547-eee61d80-5194-11eb-81f8-92ac670bf2df.png">

**Comment by davidhunter08** [comment]
Feedback from @36degrees:

> I think the issue is that your focus style is inset, so the pixels that ‘change’ change from white to yellow, not blue to yellow. So you need to be able to perceive that change. The GOV.UK and Design System search boxes both ‘grow’ so at least some pixels change from black or blue to yellow – although I wouldn’t consider either of them particularly good focus states to be honest, there’s definitely room for improvement.

> The one in the header on GOV.UK is probably better than either of them

![Screenshot 2021-01-08 at 09 48 18](https://user-images.githubusercontent.com/14331000/104003157-f6f48c00-5199-11eb-8530-266c19a9905b.png)

> There are some new success criteria in the draft for WCAG 2.2 that provide much better ‘rules’ for what make a focus state visible that are well worth a read – https://www.w3.org/TR/WCAG22/#focus-appearance-minimum

> Thinking of it in terms of ‘contrast of the change of the focus indicator’ rather than ‘contrast against the background’ helps I think

**Comment by davidhunter08** [comment]
Here's a possible solution that includes having the yellow focus outside the search box with an additional black border. 

![header-focus](https://user-images.githubusercontent.com/14331000/104909687-236d8c80-5980-11eb-8d2b-9350fd3a83eb.gif)

**Comment by idiosyn** [comment]
What about the style used on mobile? Couldn't it be used?
